cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -535,6 +541,166 @@ public void programSpecification(HttpRequest request, HttpResponder responder,
+                                String appVersion, String programType, String program, String scheduleName)
+    throws IOException, BadRequestException, NotFoundException, SchedulerException {
+    ApplicationId applicationId = new ApplicationId(namespaceId, appId, appVersion);
+    ProgramId programId = new ProgramId(applicationId, ProgramType.valueOfCategoryName(programType), program);
+    ScheduleSpecification scheduleSpecFromRequest;
+    try (Reader reader = new InputStreamReader(new ChannelBufferInputStream(request.getContent()), Charsets.UTF_8)) {
+      // The schedule spec in the request body can contain only the changed information or more, but should always
+      // contain the schedule type for deserialization
+      scheduleSpecFromRequest = GSON.fromJson(reader, ScheduleSpecification.class);
+    } catch (IOException e) {
+      LOG.debug("Error reading schedule specification from request body", e);
+      throw new IOException("Error reading request body");
+    } catch (JsonSyntaxException e) {
+      throw new BadRequestException("Request body is invalid json: " + e.getMessage());
+    }
+
+    // If the schedule name is present in the request body, it should match the name in path params
+    if (!scheduleName.equals(scheduleSpecFromRequest.getSchedule().getName())) {
+      throw new BadRequestException(
+        String.format(
+          "schedule name in the body of the request (%s) does not match the schedule name in the path parameter (%s)",
+          scheduleSpecFromRequest.getSchedule().getName(), scheduleName));
+    }
+
+    // Add program information to the schedule spec from the request
+    ScheduleSpecification scheduleSpec =
+      new ScheduleSpecification(scheduleSpecFromRequest.getSchedule(),
+                                new ScheduleProgramInfo(programId.getType().getSchedulableType(), program),
+                                scheduleSpecFromRequest.getProperties());
+    lifecycleService.updateSchedule(programId, scheduleSpec);
+    responder.sendStatus(HttpResponseStatus.OK);
+  }
+
+  @DELETE
+  @Path("apps/{app-name}/{program-type}/{program-name}/schedules/{schedule-name}")
+  public void deleteSchedule(HttpRequest request, HttpResponder responder,
+                             @PathParam("namespace-id") String namespaceId,
+                             @PathParam("app-name") String appName,
+                             @PathParam("program-type") String programType,
+                             @PathParam("program-name") String programName,
+                             @PathParam("schedule-name") String scheduleName)
+    throws NotFoundException, SchedulerException {
+    doDeleteSchedule(responder, namespaceId, appName, ApplicationId.DEFAULT_VERSION, programType, programName,
+                     scheduleName);
+  }
+
+  @DELETE
+  @Path("apps/{app-name}/versions/{app-version}/{program-type}/{program-name}/schedules/{schedule-name}")
```

Comment:
This path is inconsistent with the old paths.

`DELETE /v3/namespaces/default/apps/<app-id>/workflows/<workflow-id>/schedules/<schedule-id>` is the current path but the endpoint to resume a schedule is:

`POST /v3/namespaces/default/apps/<app-id>/versions/<app-version>/schedules/<schedule-id>/resume`

And the endpoint to suspend a schedule is:
`POST /v3/namespaces/default/apps/<app-id>/versions/<app-version>/schedules/<schedule-id>/suspend`